api/src/main/java/org/xnio/Xnio.java

```diff
@@ -74,6 +74,8 @@
 
     private static final RuntimePermission ALLOW_BLOCKING_SETTING = new RuntimePermission("changeThreadBlockingSetting");
 
+    private static final boolean MBEAN_REGISTRATION_DISABLED = Boolean.getBoolean("org.xnio.MBEAN_REGISTRATION_DISABLED");
```

Comment:
This would need to be done with a doPrivileged() block (lambda is OK).  Also the naming convention for these kinds of things is lowercase and hyphens e.g. ```org.xnio.mbean.registration-disabled```.